### PR TITLE
Carlos add hours by team member visualization project reports

### PR DESCRIPTION
**Pull request by cgomezhub**
(PRIORITY HIGH) Jae: Add “Hours by Team Member” Visualization to the Projects Reports page  (WIP Carlos)

## Related PRS (if any): 

Previous PR1960
This current PR2310 is related to Backend #[PR972](https://github.com/OneCommunityGlobal/HGNRest/pull/972)  

# Description
The graph currently displays weekly committed hours by project member. However, it does not show the actual hours worked by member on each project.
 

## Main changes explained:
- in  getTimeEntriesForSpecifiedProjectadd,  add firstName lastName  and isActive properties in 'TimeEntry' Schema from 'userProfile' Schema with will be received in the fronted.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as Owner, manager or admin user
5. Go to dashboard => report => type the project  => PieChart  => pick the charts.
6. check and test the pie charts as required 

## Note:
1.  You must add a project for at least one user ( set as a  poject member)
2. You must log hours for this member(s) in this project.
3. Go to dashboard => report => type the project  => PieChart  => pick the charts
4.See videos bellow for more info.

## Requested job:

task

## video before the change:
https://www.loom.com/share/7be14c4aec464c5fbb9669a11c4d63b5?sid=586d7ba2-8f94-4a4a-99e6-6725b31d4236

## video after the change:
https://www.loom.com/share/478133a664ff48cd832d86abdef3edd4?sid=3222bf49-79f9-4fd8-b1d8-abe4bb0687b2

## video after the change (toggle type)
https://www.loom.com/share/b4c44576df94464996eb0ef2321c4528?sid=7a1ebf85-e192-451a-b39b-1de04a6cb891

## Note:
1.  You must add a project for at least one user ( set as a  poject member)
2. You must log hours for this member(s) in this project.
3. Go to dashboard => report => type the project  => PieChart  => pick the charts
4.See videos above for more info

### CODE REVIEW

**Review by Sandhya1236 — DISMISSED** [review]
I have successfully tested the implementation of the "Hours by Team Member" visualization on the Projects Reports page. The main changes involved adding the firstName, lastName, and isActive properties to the TimeEntry schema from the userProfile schema, ensuring this data is available in the frontend. After checking into the current branch and running npm install, I cleared the site data and cache. Logged in as an Owner, Manager, or Admin user, I navigated to the dashboard, accessed the Reports section, selected a project, and viewed the PieChart. The pie charts correctly displayed the required information. Additionally, I ensured that projects had members with logged hours, verifying that the visualization worked as expected. The functionality performed seamlessly, reflecting the correct hours by team members on each project.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/79351059/13de48c8-e186-4d7a-90a3-f5e6ec040fb5

**Review by peter6866 — DISMISSED** [review]
Great work on implementing the "Hours by Team Member" visualization feature for the Projects Reports page! 
Here's the review based on the provided testing steps:

✅ Visualization Implementation:
   - Logged in as an Owner, manager, or admin user and navigated to the Projects Reports page (Dashboard => Reports => Type the project => PieChart => Pick the charts).
   - Tested the newly added "Hours by Team Member" visualization and confirmed that it displays the actual hours worked by each team member on the selected project.

✅ Data Prerequisites:
   - Acknowledged the provided notes and prerequisites for testing the feature effectively:
     1. Added at least one project with assigned team members.
     2. Logged hours for the assigned team members in the respective project.
   - Verified that the visualization accurately reflects the logged hours for the team members within the specified project.

✅ Visual Representation:
   - Reviewed the provided videos for additional information and guidance on testing the "Hours by Team Member" visualization.
   - Confirmed that the visual representation of the data aligns with the expected behavior and requirements.

https://www.loom.com/share/27a21970cb8740cf8e8bdb264f535daa?sid=ed5b6f59-94e8-4788-9333-bc372fae1863

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I tested your PR and it works as intended. Please refer the screenshot attached.
Screenshot 2024-06-01 at 1 19 28 PM

**Review by Kavil-Jain-514 — DISMISSED** [review]
The functionality is working as expected.

image

**Review by Logeshwari-Renu — DISMISSED** [review]
I have tested the PR and verified the correct data visualization of "Hours by Team Member" in the project. The chart can be easily toggled to view the Individual contribution in a project with total hours details in center. The active and Inactive members chart options can be easily viewed. The overall functionality works as intended.

PR2310
PR2310_pic2

File: src/components/Reports/ProjectReport/PiechartByProject/PieChartByProject.css [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+.pie-chart-title{
+  text-align: left;
+  margin-bottom: 24px;
+
+}
+.pie-chart-description{
```
Comment (Sudershhh) [inline src/components/Reports/ProjectReport/PiechartByProject/PieChartByProject.css:7]: The user name seems to be chipped off for bigger screen sizes. 

PR 2310

Making the text size relative to the circle would be a good start.

**Review by Sudershhh — CHANGES_REQUESTED** [review]
The functionality is working as expected.

PR 2310

**Review by jinxyou — DISMISSED** [review]
Hi I tested the case following the steps given in the instructions. All the function and pie chart work as expected. Great work!
Screenshot 2024-06-15 194516

File: src/components/Reports/ProjectReport/PiechartByProject/PieChartByProject.css (hunk 1 above)
Comment (cgomezhub) [reply src/components/Reports/ProjectReport/PiechartByProject/PieChartByProject.css:7]: Hello,
Thanks for your review, 
Please check it again

**Review by suaniii — CHANGES_REQUESTED** [review]
Hi, I've encountered an issue while reviewing your PR that causes the page to turn blank during loading. It appears that the error occurs in the renderFilters function when attempting to call "map" on an undefined value. I've attached a screenshot video for your reference.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113208952/411e89d0-0ee7-4453-91ef-e69c81b0ba21

it has been fixed, please check it out

**Comment by cgomezhub** [comment]
> The functionality is working as expected.
> 
> PR 2310

If all is good, please change the comment. Thanks

**Review by shradac — CHANGES_REQUESTED** [review]
Hey @cgomezhub, I wanted to let you know that I feature worked fine when I clicked on "Weekly Commited Hours By Active Member(Hide Piechart)" but I encountered an issue that causes the page to turn blank when clicked on "All-Time Total Hours by Member (Show PieChart". I have attached a screenshot with the error.

Screenshot 2024-06-19 at 10 04 30 PM
Screenshot 2024-06-19 at 10 09 00 PM

**Review by Sudershhh — APPROVED** [review]
I reviewed the changes, The display text seems to be relative now. 

Good job!

PR 2310 + 972

**Review by suaniii — APPROVED** [review]
It works well now.

image

**Review by jinxyou — APPROVED** [review]

**Review by Logeshwari-Renu — APPROVED** [review]
I tested the PR and checked the "Hours by Team Member" visualization in the project. The chart lets you switch views to see each person's contribution and total hours in the center. You can easily see charts for active and inactive members. Everything works as expected.

**Review by shradac — APPROVED** [review]
It works as expected now! Thanks for fixing it.

**Review by deepthikannan — APPROVED** [review]
image
Tested your PR, it's fixed and working as expected.

**Review by vishavk1992 — APPROVED** [review]
I have tested this PR it works as expected. 
PR-2310+972

**Review by Sandhya1236 — APPROVED** [review]
I have successfully tested the implementation of the "Hours by Team Member" visualization on the Projects Reports page. The main changes involved adding the firstName, lastName, and isActive properties to the TimeEntry schema from the userProfile schema, ensuring this data is available in the frontend. After checking into the current branch and running npm install, I cleared the site data and cache. Logged in as an Owner, Manager, or Admin user, I navigated to the dashboard, accessed the Reports section, selected a project, and viewed the PieChart. The pie charts correctly displayed the required information. Additionally, I ensured that projects had members with logged hours, verifying that the visualization worked as expected. The functionality performed seamlessly, reflecting the correct hours by team members on each project.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/79351059/13de48c8-e186-4d7a-90a3-f5e6ec040fb5

**Comment by sahithidhulipala9** [comment]
left a comment on backend PR 972

**Review by sahithidhulipala9 — APPROVED** [review]

**Review by Kavil-Jain-514 — APPROVED** [review]
This PR works great.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**